src/app/caesar-portal/common/services/group.service.ts

```diff
@@ -3,19 +3,58 @@ import { Http, Response } from '@angular/http';
 import { environment } from '../../../../environments/environment';
 import { Group } from '../../common/models/group';
 import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/toPromise';
 
 @Injectable()
 
 export class GroupService {
+  group: Group = new Group();
+
   constructor(private http: Http) { }
 
   getAll(): Observable<Group[]> {
     return this.http.get(environment.serviceApi.groupsUrl).
       map(result => result.json().map(obj => this.extractGroupsData(obj)));
   }
 
-  get (id: number) {
-    return this.http.get(`${environment.serviceApi.groupsUrl}/${id}`);
+  get (id: number): Promise<Group> {
+    return this.http.get(`${environment.serviceApi.groupsUrl}/${id}`).
+      toPromise().then(result => {
```

Comment:
1) toPromise? No, remove it all and use Observables
2) This is groupService and on get you should return Observable for group info and only for it
3) All other info should be loaded on the pages that needs it (IE no need to load list of teachers if you won't show it)